server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/R4237DefaultRuleApplyingMigrate.java

```diff
@@ -0,0 +1,155 @@
+        ignoreChecksum = true)
+public class R4237DefaultRuleApplyingMigrate implements JdbcMigratable {
+    private static final String MIGRATE_CONFIG_FILE = "init-config/init/regulation-rule-applying.yaml";
+
+    private DefaultRuleApplyingRepository defaultRuleApplyingRepository;
+    private RuleMetadataRepository ruleMetadataRepository;
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public void migrate(DataSource dataSource) {
+        this.defaultRuleApplyingRepository = SpringContextUtil.getBean(DefaultRuleApplyingRepository.class);
+        this.ruleMetadataRepository = SpringContextUtil.getBean(RuleMetadataRepository.class);
+        Map<String, MetadataEntity> metadataName2Metadata =
+                ruleMetadataRepository.findAll().stream().collect(Collectors.toMap(MetadataEntity::getName, e -> e));
+        List<InnerDefaultRuleApplying> expected =
+                YamlUtils.fromYaml(MIGRATE_CONFIG_FILE, new TypeReference<List<InnerDefaultRuleApplying>>() {});
+        Map<String, List<DefaultRuleApplyingEntity>> actualRulesetName2RuleApplyings = defaultRuleApplyingRepository
+                .findAll().stream().collect(Collectors.groupingBy(DefaultRuleApplyingEntity::getRulesetName));
+        List<DefaultRuleApplyingEntity> toAdd = new ArrayList<>();
+        for (InnerDefaultRuleApplying expectedApplying : expected) {
+            String rulesetName = expectedApplying.getRulesetName();
+            MetadataEntity metadataEntity = metadataName2Metadata.get(expectedApplying.getRuleName());
+            if (Objects.isNull(metadataEntity)) {
+                throw new UnexpectedException("rule metadata not found, ruleName: " + expectedApplying.getRuleName());
+            }
+
+            List<DefaultRuleApplyingEntity> actualRuleApplyings =
+                    actualRulesetName2RuleApplyings.get(rulesetName);
+            if (CollectionUtils.isEmpty(actualRuleApplyings)) {
+                toAdd.add(generateNewEntity(expectedApplying, rulesetName, metadataEntity.getId()));
+            } else {
+                Optional<DefaultRuleApplyingEntity> existed = actualRuleApplyings.stream()
+                        .filter(r -> metadataEntity.getId() == r.getRuleMetadataId()).findFirst();
+                if (existed.isPresent()) {
+                    DefaultRuleApplyingEntity actualRuleApplying = existed.get();
+                    if (!isApplyingEquals(expectedApplying, actualRuleApplying)) {
+                        actualRuleApplying.setEnabled(expectedApplying.getEnabled());
+                        actualRuleApplying.setLevel(expectedApplying.getLevel());
+                        actualRuleApplying.setAppliedDialectTypes(expectedApplying.getAppliedDialectTypes());
+                        actualRuleApplying.setPropertiesJson(expectedApplying.getPropertiesJson());
+                        toAdd.add(actualRuleApplying);
```

Comment:
how to prevent unique constraint of table 'regulation_default_rule_applying'?

Reply:
` SimpleRepository#saveAll` could be used in both `insert` and `update` scenarios. If the `@Id` existed in the table, then it will update the record; otherwise it insert a new record.